src/server/game/Spells/Spell.cpp

```diff
@@ -3010,8 +3009,7 @@ SpellMissInfo Spell::DoSpellHitOnUnit(Unit* unit, uint32 effectMask, bool scaleA
             unit->IncrDiminishing(m_diminishGroup);
     }
 
-    // Xinef: Stealth remove, added UGLY hack for mass dispel... cant find any other solution, ugly hack for Storm, Earth and Fire talent
-    if (m_caster != unit && m_caster->IsHostileTo(unit) && !m_spellInfo->IsPositive() && !m_triggeredByAuraSpell && m_spellInfo->SpellIconID != 2267 && unit->IsControlledByPlayer() && (!m_caster->IsTotem() || m_spellInfo->Id == 64695) && /*Intervene Trigger*/ m_spellInfo->Id != 59667)
+    if (!m_spellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_BREAK_STEALTH))
```

Comment:
shouldn't this be
```cpp
if (!m_spellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_BREAK_STEALTH) && !m_spellInfo->HasAttribute(SPELL_ATTR1_NOT_BREAK_STEALTH));
```

Or maybe that is handled here https://github.com/azerothcore/azerothcore-wotlk/blob/1a802daf0fc3178b3d21402e2884c8266656f081/src/server/game/Spells/SpellInfo.cpp#L1240 or somewhere else, not really sure